cpp/src/tsne/barnes_hut_tsne.cuh

```diff
@@ -310,6 +310,14 @@ value_t Barnes_Hut(value_t* VAL,
                                                                      n);
     CUDA_CHECK(cudaPeekAtLastError());
 
+    value_t grad_norm =
+      compute_grad_norm(handle, old_forces.data(), old_forces.size());
+
+    if (grad_norm <= params.min_grad_norm) {
```

Comment:
Probably need to delete line 78 of tsne.h.

https://github.com/rapidsai/cuml/blob/c364d051df55c16473dac9ba89db752f8723b180/cpp/include/cuml/manifold/tsne.h#L77-L79